src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DateTimeTest/DateTimeTest.cs

```diff
@@ -504,6 +520,249 @@ public static void ReaderParameterTest()
+                            p2.Value = DBNull.Value;
+
+                            cmd2.ExecuteNonQuery();
+                            pi.Value = 1;
+                            p1.Value = new DateOnly(2000, 12, 31);
+                            p2.Value = new TimeOnly(23, 59, 59);
+                            cmd2.ExecuteNonQuery();
+
+                            // Test 2
+                            cmd2.CommandText = "SELECT COUNT(*) FROM " + tableName + " WHERE @pi = ci AND @p1 = c1 AND @p2 = c2";
+                            pi.Value = 0;
+                            p1.Value = new DateOnly(1753, 1, 1);
+                            p2.Value = new TimeOnly(20, 12, 13, 360);
+                            object scalarResult = cmd2.ExecuteScalar();
+                            Assert.True(scalarResult.Equals(1), string.Format("FAILED: Execute scalar returned unexpected result. Expected: {0}. Actual: {1}.", 1, scalarResult));
+
+                            cmd2.Parameters.Clear();
+                            pi = cmd2.Parameters.Add("@pi", SqlDbType.Int);
+                            p1 = cmd2.Parameters.Add("@p1", SqlDbType.Date);
+                            p2 = cmd2.Parameters.Add("@p2", SqlDbType.Time);
+                            pi.SqlValue = new SqlInt32(0);
+                            p1.SqlValue = new DateOnly(1753, 1, 1);
+                            p2.SqlValue = new TimeOnly(20, 12, 13, 360);
+                            p2.Scale = 3;
+                            scalarResult = cmd2.ExecuteScalar();
+                            Assert.True(scalarResult.Equals(1), string.Format("FAILED: ExecutScalar returned unexpected result. Expected: {0}. Actual: {1}.", 1, scalarResult));
+
+                            // Test 3
+
+                            cmd.CommandText = createProc;
+                            cmd.ExecuteNonQuery();
+                            cmd.CommandText = createProcN;
+                            cmd.ExecuteNonQuery();
+                            using (SqlCommand cmd3 = conn.CreateCommand())
+                            {
+                                cmd3.CommandType = CommandType.StoredProcedure;
+                                cmd3.CommandText = procName;
+                                p1 = cmd3.Parameters.Add("@p1", SqlDbType.Date);
+                                p2 = cmd3.Parameters.Add("@p2", SqlDbType.Time);
+                                p1.Direction = ParameterDirection.Output;
+                                p2.Direction = ParameterDirection.Output;
+                                p2.Scale = 7;
+                                cmd3.ExecuteNonQuery();
+
+                                Assert.True(p1.Value.Equals(new DateTime(1753, 1, 1)), "FAILED: SqlParameter p1 contained incorrect value");
+                                Assert.True(p2.Value.Equals(new TimeSpan(0, 20, 12, 13, 360)), "FAILED: SqlParameter p2 contained incorrect value");
```

Comment:
Assume this is expected?

Reply:
Hard to tell, any more context?

Reply:
If the retrun value was TimeOnly it would be very breaking, so yes